frontend/src/metabase/public/lib/code.ts

```diff
@@ -0,0 +1,86 @@
+import type {
+  ClientCodeSampleConfig,
+  CodeSampleParameters,
+  ServerCodeSampleConfig,
+} from "./types";
+import {
+  clojure,
+  getHtmlSource,
+  getJsxSource,
+  getPugSource,
+  node,
+  python,
+  ruby,
+} from "./code-templates";
+
+export const getEmbedClientCodeExampleOptions =
+  (): ClientCodeSampleConfig[] => [
+    {
+      name: "Pug / Jade",
+      source: getPugSource({ iframeUrl: `iframeUrl` }),
+      mode: "ace/mode/jade",
+    },
+    {
+      name: "Mustache",
+      source: getHtmlSource({ iframeUrl: `"{{iframeUrl}}"` }),
+      mode: "ace/mode/html",
+    },
+    {
+      name: "ERB",
+      source: getHtmlSource({ iframeUrl: `"<%= @iframe_url %>"` }),
+      mode: "ace/mode/html_ruby",
+    },
+    {
+      name: "JSX",
+      source: getJsxSource({ iframeUrl: `{iframeUrl}` }),
+      mode: "ace/mode/jsx",
+    },
+  ];
+
+export const getEmbedServerCodeExampleOptions = (
+  codeSampleParameters: CodeSampleParameters,
+): ServerCodeSampleConfig[] => [
+  {
+    name: "Node.js",
+    source: node.getServerSource(codeSampleParameters),
+    parametersSource: node.getParametersSource(codeSampleParameters.params),
+    getIframeQuerySource: node.getIframeQuerySource(
+      codeSampleParameters.displayOptions,
+    ),
+    mode: "ace/mode/javascript",
+    embedOption: "Pug / Jade",
```

Comment:
Are all these options for the given server language finalized? I kind of remember the previous behavior, we only change the template language to ERB for Ruby, but not others? I could be wrong though. Could you double-check this?